python/ray/tests/test_actor_failures.py

```diff
@@ -204,6 +204,45 @@ def get_pid(self):
         timeout_ms=5000)
 
 
+def test_actor_caller_reconstruction(ray_start_regular):
+    """Test the caller of an actor is killed and then reconstructed."""
+
+    @ray.remote(max_reconstructions=1)
+    class ReconstructableActor:
+        """An actor that will be reconstructed at most once."""
+
+        def __init__(self, actor):
+            self.actor = actor
+
+        def increase(self):
+            return ray.get(self.actor.increase.remote())
+
+        def get_pid(self):
+            return os.getpid()
+    
+    @ray.remote(max_reconstructions=1)
+    class Actor:
+        """An actor that will be reconstructed at most once."""
+
+        def __init__(self):
+            self.value = 0
+
+        def increase(self):
+            self.value += 1
+            return self.value
+
+    remote_actor = Actor.remote()
+    actor = ReconstructableActor.remote(remote_actor)
```

Comment:
Would also probably be good to test that this works when the actor is created in a different process/node

Reply:
There shouldn't be a difference in behavior for this PR, whether the reconstructed actor is on a remote node or local node. But if you think it would be useful, I can add a case.